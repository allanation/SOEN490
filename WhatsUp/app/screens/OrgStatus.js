import {
  StyleSheet,
  View,
  ScrollView,
  Platform,
  Text,
} from "react-native";
import React from "react";
import Screen from "../components/Screen";
import PropTypes from "prop-types";
import { format } from "date-fns";

OrgStatus.propTypes = {
  route: PropTypes.any,
};

export const convertTime = (number) => {
  return number ? format(new Date(number), "hh:mm aaaaa'm'") : "";
};

function OrgStatus({ route }) {
  const { prop } = route.params;

  return (
    <Screen style={{ padding: "5%", backgroundColor: "white" }}>
      <ScrollView style={{ width: "100%", display: "flex" }}>
        <View>
          <Text
            style={{
              marginBottom: "2%",
              fontSize: 18,
              fontWeight: "bold",
              marginLeft: "1%",
            }}
          >
            Status
          </Text>
        </View>
        <View>
          <Text style={styles.text}>{prop.eventStatus}</Text>
        </View>
        <View>
          <Text
            style={{
              marginBottom: "2%",
              fontSize: 18,
              fontWeight: "bold",
              marginLeft: "1%",
              marginTop: "7.5%",
            }}
          >
            {prop.eventStatus === "Approved" ? "Number of Participants" : ""}
          </Text>
        </View>
        <View>
          <Text style={styles.text}>
            {prop.eventStatus === "Approved" ? prop.numberOfParticipants : ""}
          </Text>
        </View>
        <View>
          <Text
            style={{
              marginBottom: "2%",
              fontSize: 18,
              fontWeight: "bold",
              marginLeft: "1%",
              marginTop: "7.5%",
            }}
          >
            {prop.eventStatus === "Approved" ? "Number of Favorites" : ""}
          </Text>
        </View>
        <View>
          <Text style={styles.text}>
            {prop.eventStatus === "Approved" ? prop.numberOfFavorites : ""}
          </Text>
        </View>
        <View>
          <Text
            style={{
              marginBottom: "2%",
              fontSize: 18,
              fontWeight: "bold",
              marginLeft: "1%",
              marginTop: "7.5%",
            }}
          >
            {prop.eventStatus === "Rejected" ? "Comments from Admin" : ""}
          </Text>
        </View>
        <View>
          <Text style={styles.text}>
            {prop.eventStatus === "Rejected" ? prop.adminComment : ""}
          </Text>
        </View>
      </ScrollView>
    </Screen>
  );
}

const styles = StyleSheet.create({
  text: {
    color: "#100101",
    fontSize: 16,
    fontFamily: Platform.OS === "ios" ? "Helvetica Neue" : "sans-serif",
    marginTop: "2%",
    marginBottom: "2%",
    marginLeft: "1%",
  },
});

export default OrgStatus;
