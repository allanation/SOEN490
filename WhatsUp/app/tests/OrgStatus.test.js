import "react-native";
import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import {
  render,
} from "@testing-library/react-native";
import OrgStatus from "../screens/OrgStatus";

jest.useFakeTimers();

it("Renders correctly organizer status page", async () => {
    const tree = render(
      <NavigationContainer>
        <OrgStatus />
      </NavigationContainer>
    ).toJSON();
  
    expect(tree).toMatchSnapshot();
  });