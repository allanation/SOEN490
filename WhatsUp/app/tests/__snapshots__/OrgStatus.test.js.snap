// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly organizer status page 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {
        "backgroundColor": "white",
        "padding": "5%",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTScrollView
    style={
      Object {
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": "2%",
              "marginLeft": "1%",
            }
          }
        >
          Status
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#100101",
              "fontFamily": "Helvetica Neue",
              "fontSize": 16,
              "marginBottom": "2%",
              "marginLeft": "1%",
              "marginTop": "2%",
            }
          }
        >
          Rejected
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": "2%",
              "marginLeft": "1%",
              "marginTop": "7.5%",
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#100101",
              "fontFamily": "Helvetica Neue",
              "fontSize": 16,
              "marginBottom": "2%",
              "marginLeft": "1%",
              "marginTop": "2%",
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": "2%",
              "marginLeft": "1%",
              "marginTop": "7.5%",
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#100101",
              "fontFamily": "Helvetica Neue",
              "fontSize": 16,
              "marginBottom": "2%",
              "marginLeft": "1%",
              "marginTop": "2%",
            }
          }
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": "2%",
              "marginLeft": "1%",
              "marginTop": "7.5%",
            }
          }
        >
          Comments from Admin
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#100101",
              "fontFamily": "Helvetica Neue",
              "fontSize": 16,
              "marginBottom": "2%",
              "marginLeft": "1%",
              "marginTop": "2%",
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
